### PR TITLE
Add stubs for BGP plugin

### PR DESCRIPTION
**Pull request by jwbensley**
This PR adds mypy stubs for the NetBox [BGP plugin](https://github.com/k01ek/netbox-bgp).

### CODE REVIEW

File: pynetbox-stubs/core/api.pyi [hunk 1]
```diff
@@ -9,10 +9,7 @@ from pynetbox._gen.tenancy import TenancyApp
 from pynetbox._gen.users import UsersApp
 from pynetbox._gen.virtualization import VirtualizationApp
 from pynetbox._gen.wireless import WirelessApp
-from pynetbox.core.app import App
-
-class PluginsApp:
-    pass
+from pynetbox.core.app import App, PluginsApp
```
Comment (sebastian-philipp) [inline pynetbox-stubs/core/api.pyi:13]: missing empty line: did you run `tox -e fixlint`?

File: pynetbox-stubs/core/app.pyi [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 
 class App:
     def __init__(self, api: "Api", name: str):
         self.api = api
         self.name = name
+
+class PluginsApp:
+    def __init__(self, api: "Api"):
+        self.api = api
+        self.bgp = BgpApp(self.api, "bgp")
```
Comment (sebastian-philipp) [inline pynetbox-stubs/core/app.pyi:16]: Very good question. This means, everyone who want to use this repository needs their own generated stub files. 

You think we can avoid that by somehow proving a generic "catches all" variant here? See https://github.com/netbox-community/pynetbox/issues/333 for the upstream issue.

File: pynetbox-stubs/core/api.pyi (hunk 1 above)
Comment (jwbensley) [reply pynetbox-stubs/core/api.pyi:13]: `missing empty line: did you run tox -e fixlint?`

tox is throwing loads of errors about all the other files in the repo (files I haven't touched), specifically black and isort, so it was tricky to filter out that stuff. I didn't want to raise a PR with changes to those other files (I will do this in a separate PR though)... So it seems I missed this in all the tox noise. Have fixed now.

Comment (sebastian-philipp) [reply pynetbox-stubs/core/api.pyi:13]: interesting. thanks for this

File: pynetbox-stubs/core/app.pyi (hunk 2 above)
Comment (jwbensley) [reply pynetbox-stubs/core/app.pyi:17]: Added the override of `__getattr__()`.

**Pull request merged**